runtime/src/main/java/org/corfudb/runtime/view/stream/AddressMapStreamView.java

```diff
@@ -394,6 +394,15 @@ protected long getMaxGlobalFromMaxEntries(int maxEntries) {
         StreamAddressSpace streamAddressSpace = runtime.getSequencerView()
                 .getStreamAddressSpace(new StreamAddressRange(getId(), Address.MAX, getCurrentGlobalPosition()));
 
+        // Validate the streaming start address (seeked address) is beyond the trim mark, otherwise, updates would be missed
+        // (as there is no further resolution for this space).
+        // A TrimmedException indicates a snapshot needs to be performed to sync the object from its checkpoint, followed
+        // by a re-subscription from a point beyond the trimmed address (i.e., the snapshot address)
+        if (getCurrentGlobalPosition() < streamAddressSpace.getTrimMark()) {
+            throw new TrimmedException(String.format("Stream seeked to %s, but address space is trimmed at %s", getCurrentGlobalPosition(),
+                    streamAddressSpace.getTrimMark()));
+        }
+
```

Comment:
Why isn't this exception discovered during `remainingUpTo` ?

Reply:
The stream being polled is opened with

```java

        StreamOptions options = StreamOptions
                .builder()
                .cacheEntries(false)
                .isCheckpointCapable(false)
                .build();

```

This should be sufficient to detect and throw a trimmed exception. If it's not, then there might be a bug.

Reply:
Good catch, actually the bug is really returning Address.NON_ADDRESS when the address space size == 0. Because my test case happened to fall in this case it revealed the bug. If the log would've been partially trimmed we would've observed the TrimmedException. I did the correct fix and added a test case such that we cover both scenarios (partial trim and full trim)